Assets/Project/Modules/VFX/Generic/Scripts/ParticleBehaviours/MaterialInterpolatorRecycleParticle.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using DG.Tweening;
+using Popeye.Core.Pool;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace Popeye.Modules.VFX.Generic.ParticleBehaviours
+{
+    public class MaterialInterpolatorRecycleParticle : RecyclableObject
+    {
+        [SerializeField] private MeshRenderer _meshRenderer;
+        private Material _material;
+
+        [System.Serializable]
+        private class FloatSetupData
+        {
+            public string id;
+            public float initialValue;
+        }
+        
+        [System.Serializable]
+        private class FloatInterpolationData
+        {
+            public float delay;
+            
+            public string id;
+            public float duration;
+            public float endValue = 1.0f;
+            public Ease ease;
+            
+            public bool waitForCompletion;
+        }
```

Comment:
I think that both this classes should be unified into one alone.
The variables should be private + [SerializeField], with lambda properties (=>) for access only.

Also, make it a ScriptableObject.

Reply:
If you want to optimize the material property access, you can do: 
`propertyId = Shader.PropertyToID(propertyName)`
in the ScriptableObjects's Awake()

Reply:
I do not see the advantages of merging these classes, both serve a different purpose and need to be iterated in different time and order. When it comes to the scriptable object, wouldn't it just make it harder to edit and create? Since it would require extra steps with no benefits.